### PR TITLE
[WFLY-19833] Refactor the jaxrs subsystem to use the newer subsystem API's

### PR DESCRIPTION
**Pull request by jamezp**
https://issues.redhat.com/browse/WFLY-19833

I attempted to put these into separate logical commits. These could be squashed, but it seems it will be easier to review with separate commits.

### CODE REVIEW

**Comment by bstansberry** [comment]
@jamezp Please sort the https://github.com/wildfly/wildfly-proposals/pull/367 discussion with @jimma and if the RFE is going first please put a 36.x label on this. Otherwise please make noise on zulip to ensure this gets in ASAP and unblocks Jim.

If Jim's is going first and gets done in plenty of time perhaps 36.x label can come off, but we'll be busy in December.

**Comment by jamezp** [comment]
@bstansberry Yes. I think @pferraro is going to have some time to review this tomorrow. Once this is resolved, it will make adding new attributes much easier. Plus we'll have an easier way to write tests for new attributes.

**Review by pferraro — CHANGES_REQUESTED** [review]

File: jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsAttribute.java [hunk 1]
```diff
@@ -36,6 +43,79 @@ public abstract class JaxrsAttribute {
 
     public static final String RESTEASY_PARAMETER_GROUP = "resteasy";
 
+    private static final AttributeParser ELEMENT_PARSER = new AttributeParser() {
```
[the remark this answers is not in the document]
Comment (jamezp) [reply jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsAttribute.java:46]: I will test that. I was looking for one and may have just missed it. Thank you!

**Review by pferraro — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jamezp and @pferraro

**Pull request merged**